openreview/conference/invitation.py

```diff
@@ -600,27 +605,144 @@ def set_recommendation_invitation(self, conference, start_date, due_date, notes_
+                    }
+                }
             paper_recommendation_invitation = openreview.Invitation(
-                id = conference.get_id() + '/-/Paper{number}/Recommendation'.format(number = note.number),
+                id = conference.get_recommendation_id(number = note.number),
+                cdate = tools.datetime_millis(start_date),
+                duedate = tools.datetime_millis(due_date),
+                expdate = tools.datetime_millis(due_date + datetime.timedelta(minutes = SHORT_BUFFER_MIN)),
                 super = recommendation_invitation.id,
                 invitees = [conference.get_program_chairs_id(), conference.get_id() + '/Paper{number}/Area_Chairs'.format(number = note.number)],
                 writers = [conference.get_id()],
                 signatures = [conference.get_id()],
                 multiReply = True,
-                reply = {
-                    'forum': note.id,
-                    'content': {
-                        'tag': {
-                            'description': 'Recommend reviewer',
-                            'order': 1,
-                            'required': True,
-                            'values-dropdown': reviewers
-                        }
-                    }
-                }
+                reply = reply
             )
             paper_recommendation_invitation = self.client.post_invitation(paper_recommendation_invitation)
-            print('Posted', paper_recommendation_invitation.id)
+
+
+    def set_registration_invitation(self, conference, start_date, due_date, with_area_chairs):
+
+        invitees = []
+        if with_area_chairs:
+            invitees.append(conference.get_area_chairs_id())
+        invitees.append(conference.get_reviewers_id())
+
+        subj_desc = 'To properly assign papers to reviewers, we ask that reviewers provide their areas of expertise from among the provided list of subject areas. Please submit your areas of expertise by selecting the appropriate options from the "Subject Areas" list.\n\n'
+
+        coi_desc = 'In order to avoid conflicts of interest in reviewing, we ask that all reviewers take a moment to update their OpenReview profiles with their latest information regarding work history and professional relationships. After you have updated your profile, please confirm that your OpenReview profile is up-to-date by selecting yes in the "Profile Confirmed" section.\n\n'
+
+        tpms_desc = 'In addition to subject areas, we will be using the Toronto Paper Matching System (TPMS) to compute paper-reviewer affinity scores. Please take a moment to sign up for TPMS and/or update your TPMS account with your latest papers. Then, please ensure that the email address that is affiliated with your TPMS account is linked to your OpenReview profile. After you have done this, please confirm that your TPMS account is up-to-date by selecting yes in the "TPMS Account Confirmed" section.\n\n'
+
```

Comment:
what do you think about adding the data consent question here, too?

there may be good reasons to leave it out, I'm not sure. but it would be optional.

Reply:
I didn't do anything optional here. It is prepared for a conference like UAI. I will create an issue to make the registration fields configurable.

Reply:
sorry, what I mean is that consenting to have their data collected would be optional. the question itself would be there by default, so that we're always asking if we can use/publish their data.

Reply:
Yes, we can do that but in another PR. I had to add this registration phase just to calculate the subject areas scores for the matching.

Reply:
I created an issue here: https://github.com/iesl/openreview-py/issues/250

We can add more detail and I will implement it soon.